### PR TITLE
Refactor/string ids

### PR DESCRIPTION
**Pull request by rdrdrdrd95**
Open issue:

Reduce inclusion of StringIds.h in header files #741

### CODE REVIEW

File: src/OpenLoco/Window.h [hunk 1]
```diff
@@ -91,92 +90,28 @@ namespace OpenLoco::Ui
...
-        return out;
-    }
-
-    constexpr widget_t makeRemapWidget(Gfx::point_t origin, Gfx::ui_size_t size, widget_type type, uint8_t colour, uint32_t content = 0xFFFFFFFF, string_id tooltip = StringIds::null)
-    {
-        widget_t out = makeWidget(origin, size, type, colour, content, tooltip);
-
-        // TODO: implement this as a constant.
-        out.content |= (1 << 29);
+    extern constexpr widget_t makeWidget(Gfx::point_t origin, Gfx::ui_size_t size, widget_type type, uint8_t colour, uint32_t content = 0xFFFFFFFF, string_id tooltip = StringIds::null);
```
Comment (duncanspumpkin) [inline src/OpenLoco/Window.h:93]: pretty sure you cant do constexpr's extern.

**Comment by duncanspumpkin** [comment]
Unfortunately due to constexprs being like templates and inline in that they must be immediately defined you can't use extern and must have the full definition in the header. It makes this pr just the deletion of one or two #includes.

**Comment by duncanspumpkin** [comment]
The widget one definitely needs to stay constexpr the vehicle one I'm not sure. Even if VS is happy with it afaik it's not standards compliant so it will probably not work on gcc and clang.

**Comment by duncanspumpkin** [comment]
@rdrdrdrd95 I've taken this PR and sort of reverted all of the constexpr extern changes. Sorry we should have really thought through the issue before putting it up as a good first issue.

File: src/OpenLoco/Objects/CompetitorObject.h [hunk 2]
```diff
@@ -1,6 +1,4 @@
 #pragma once
-
-#include "../Localisation/StringIds.h"
 #include "../Localisation/StringManager.h"
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Objects/CompetitorObject.h:2]: Use Types.hpp instead of StringManager.h

File: src/OpenLoco/Window.h [hunk 3]
```diff
@@ -4,7 +4,6 @@
 #include "Core/Optional.hpp"
 #include "Graphics/Gfx.h"
 #include "Interop/Interop.hpp"
-#include "Localisation/StringIds.h"
 #include "Localisation/StringManager.h"
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Window.h:7]: StringManager include can be dropped

**Review by AaronVanGeffen — APPROVED** [review]

**Comment by rdrdrdrd95** [comment]
Thanks for following up on that, I have been swamped on-boarding our new college hires for the past few weeks.

**Pull request merged**